src/pigeon/http/common.py

```diff
@@ -1,17 +1,14 @@
 import pigeon.conf
+from pigeon.middleware.registry import error_handler
 from http import HTTPStatus
+from pigeon.http import HTTPResponse, HTTPRequest
 
 
-def error(code: int, request=None):
+def error(code: int, request: HTTPRequest | None = None) -> HTTPResponse | str:
     """
-    Returns the HTTPResponse for the error code provided
+    Returns the HTTPResponse for the error code provided, request parameter optional
     """
-    # if a specific error view for the error code exists
```

Comment:
fallback is a special error view, which should receive as arguments (code, request | None)